File: java/src/org/shypl/sna/SocialNetwork.java
package org.shypl.sna;

public abstract class SocialNetwork {
	private final int    id;
	private final String name;

	public SocialNetwork(int id, String name) {
		this.id = id;
		this.name = name;
	}

	public int getId() {
		return id;
	}

	public String getName() {
		return name;
	}

	@Override
	public String toString() {
		return name ;
	}

	@Override
	public boolean equals(Object obj) {
		return obj == this || obj instanceof SocialNetwork && ((SocialNetwork)obj).id == id;
	}
}

File: php/src/org/shypl/sna/SocialNetwork.php
<?php

namespace org\shypl\sna;

abstract class SocialNetwork {
	/**
	 * @var int
	 */
	private $id;
	/**
	 * @var string
	 */
	private $name;

	/**
	 * @param int    $id
	 * @param string $name
	 */
	public function __construct($id, $name) {
		$this->id = $id;
		$this->name = $name;
	}

	/**
	 * @return int
	 */
	public function getId() {
		return $this->id;
	}

	/**
	 * @return string
	 */
	public function getName() {
		return $this->name;
	}

	/**
	 * @param array $parameters
	 *
	 * @return Adapter
	 */
	public abstract function createAdapter(array $parameters);
}